src/excited_workflow/carbon_tracker.py

```diff
@@ -0,0 +1,371 @@
+"""Train carbon tracker datasets."""
+
+import datetime
+from pathlib import Path
+from typing import Any
+
+import matplotlib.pyplot as plt
+import numpy as np
+import onnxmltools
+import pandas as pd
+import pycaret.regression
+import xarray as xr
+from dask.distributed import Client
+from skl2onnx.common.data_types import DoubleTensorType
+
+import excited_workflow
+from excited_workflow.source_datasets import datasets
+
+
+def mask_region(
+    regions: Path,
+    target: Path,
+    ds_input: xr.Dataset,
+    mask: int,
+    y_key: str,
+) -> tuple[xr.Dataset, pd.DataFrame]:
+    """Limit data to a region and time slice.
+
+    Args:
+        regions: path to regions file.
+        target: path to target dataset.
+        ds_input: input dataset.
+        mask: name of region to mask to.
+        y_key: target variable name.
+
+    Returns:
+        Masked dataset and dataframe.
+    """
+    ds_regions = xr.open_dataset(regions)
+    ds_cb = xr.open_dataset(target)
+    ds_cb = excited_workflow.utils.convert_timestamps(ds_cb)
+    ds_merged = xr.merge(
+        [
+            ds_cb[[y_key]],
+            ds_regions["transcom_regions"],
+            ds_input,
+        ]
+    )
+
+    ds_sel = ds_merged.sel({"time": slice("2000-01", "2019-12")})
+    ds_sel = ds_sel.compute()
+    ds_sel = ds_sel.where(ds_merged["transcom_regions"] == mask)
+    ds_sel[y_key].attrs["long_name"] = "terrestrial biosphere CO2 flux"
+    df_sel = ds_sel.to_dataframe().dropna()
+
+    return ds_sel, df_sel
```

Comment:
I'm not a big fan of returning the same data (essentially) twice... Let's leave it for now, I see why you did it. It just feels a bit clunky.

Reply:
Yea I felt the same and tried to find another way but couldn't think of anything better for now.